service/history/api/get_workflow_util.go

```diff
@@ -238,14 +238,29 @@ func GetMutableState(
 func MutableStateToGetResponse(
 	mutableState workflow.MutableState,
 ) (*historyservice.GetMutableStateResponse, error) {
-	currentBranchToken, err := mutableState.GetCurrentBranchToken()
-	if err != nil {
+	// NOTE: fields of GetMutableStateResponse (returned value) are accessed outside of workflow lock
+	// and, therefore, MUST be copied by value from mutableState fields.
+
+	var currentBranchToken []byte
+	if cbt, err := mutableState.GetCurrentBranchToken(); err != nil {
 		return nil, err
+	} else {
+		currentBranchToken = make([]byte, len(cbt))
+		copy(currentBranchToken, cbt)
 	}
```

Comment:
I agree with the linter this time, how about
```suggestion
	currentBranchToken, err := mutableState.GetCurrentBranchToken()
	if err != nil {
		return nil, err
	}
	currentBranchToken = slices.Clone(currentBranchToken)
```
?

Reply:
but actually wait.. we don't mutate the byte slice itself, right?? sharing this one should be fine